File: src-tauri/src/models/agent_chat.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentChatSession {
    pub id: String,
    pub workspace_id: String,
    pub provider: String,
    pub status: String,
    pub title: String,
    pub provider_session_id: Option<String>,
    pub cwd: String,
    pub raw_output: String,
    pub created_at: String,
    pub updated_at: String,
    pub ended_at: Option<String>,
    pub closed_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentChatEvent {
    pub id: String,
    pub session_id: String,
    pub seq: i64,
    pub event_type: String,
    pub role: Option<String>,
    pub title: Option<String>,
    pub body: String,
    pub status: Option<String>,
    pub metadata: Option<serde_json::Value>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateAgentChatSessionInput {
    pub workspace_id: String,
    pub provider: String,
    pub title: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SendAgentChatMessageInput {
    pub session_id: String,
    pub prompt: String,
    pub profile_id: Option<String>,
    pub task_mode: Option<String>,
    pub reasoning: Option<String>,
    pub claude_agent: Option<String>,
    pub model: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentChatEventEnvelope {
    pub workspace_id: String,
    pub session: AgentChatSession,
    pub event: AgentChatEvent,
}

File: src/types/agent-chat.ts
export type AgentChatProvider = 'claude_code' | 'codex' | 'kimi_code' | string;
export type AgentChatStatus = 'idle' | 'running' | 'succeeded' | 'failed' | 'interrupted' | string;

export interface AgentChatSession {
  id: string;
  workspaceId: string;
  provider: AgentChatProvider;
  status: AgentChatStatus;
  title: string;
  providerSessionId?: string | null;
  cwd: string;
  rawOutput: string;
  createdAt: string;
  updatedAt: string;
  endedAt?: string | null;
  closedAt?: string | null;
}

export interface AgentChatEvent {
  id: string;
  sessionId: string;
  seq: number;
  eventType: 'user_message' | 'assistant_message' | 'thinking' | 'plan' | 'todo' | 'tool_call' | 'tool_result' | 'command' | 'file_read' | 'file_change' | 'test_run' | 'error' | 'result' | 'next_action' | 'diagnostic' | 'status' | string;
  role?: 'user' | 'assistant' | string | null;
  title?: string | null;
  body: string;
  status?: string | null;
  metadata?: AgentChatEventMetadata | null;
  createdAt: string;
}

export interface AgentChatNextAction {
  id: string;
  label: string;
  kind: 'accept_plan' | 'ask_followup' | 'switch_to_act' | 'copy_plan' | 'review_diff' | 'run_tests' | 'ask_reviewer' | 'create_pr' | 'open_diagnostics' | 'send_failure' | 'refresh_comments' | 'check_merge' | 'archive_chat' | string;
  tone?: 'primary' | 'muted' | 'warning' | 'danger' | string;
}

export interface AgentChatEventMetadata {
  command?: string;
  exitCode?: number | null;
  path?: string;
  paths?: string[];
  summary?: string;
  risk?: string;
  testStatus?: string;
  nextActions?: AgentChatNextAction[];
  planId?: string;
  taskMode?: string;
  claudeAgent?: string;
  model?: string;
  reasoning?: string;
  [key: string]: unknown;
}

export interface AgentChatEventEnvelope {
  workspaceId: string;
  session: AgentChatSession;
  event: AgentChatEvent;
}

export interface CreateAgentChatSessionInput {
  workspaceId: string;
  provider: AgentChatProvider;
  title?: string;
}

export interface SendAgentChatMessageInput {
  sessionId: string;
  prompt: string;
  profileId?: string;
  taskMode?: string;
  reasoning?: string;
  claudeAgent?: string;
  model?: string;
}
